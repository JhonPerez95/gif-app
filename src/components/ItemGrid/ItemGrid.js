import React from 'react';
import PropTypes from 'prop-types';

const ItemGrid = ({ id, title, url }) => {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

ItemGrid.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default ItemGrid;
