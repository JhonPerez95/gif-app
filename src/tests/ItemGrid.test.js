import React from 'react';
import ItemGrid from '../components/ItemGrid/ItemGrid';
import { shallow } from 'enzyme';

describe('Pruebas al Componente ItemGrid ', () => {
  test('Renderizado correctamente del componente ', () => {
    const title = 'Naruto';
    const url = 'https://gify.com';
    const wrapper = shallow(<ItemGrid title={title} url={url} />);
    expect(wrapper).toMatchSnapshot();
  });
});
