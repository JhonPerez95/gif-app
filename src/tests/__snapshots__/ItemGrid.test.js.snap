// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al Componente ItemGrid  Renderizado correctamente del componente  1`] = `
<div
  className="card animate__animated animate__rubberBand"
>
  <img
    alt="Naruto"
    src="https://gify.com"
  />
  <p>
    Naruto
  </p>
</div>
`;
